### ISSUE TITLE
bug on route with "/" at the end

### ISSUE THREAD

**Issue by carvalhoviniciusluiz** [restify/node-restify#1631]
Hello
every route with a / at the end without this passing a parameter throws a ResourceNotFound
```javascript
var restify = require('restify');

function respond(req, res, next) {
  res.send('hello ' + `${req.params.name || 'nameless'}`);
  next();
}

var server = restify.createServer();
server.get('/hello/:name', respond);
server.get('/hello', respond);
server.get('/hello2/', respond); // <-- here throw ResourceNotFound

server.listen(8080, function() {
  console.log('%s listening at %s', server.name, server.url);
});
```
`/hello` and `/hello2/` has the same configuration, the difference is that `/hello2/` throws an exception 
```
{
"code": "ResourceNotFound",
"message": "/hello2 does not exist"
}
```

this causes confusion if / is not intentional.

### env
Restify Version: 7.0.0
Node.js Version: 8.9.3

**Comment by hekike** [comment]
`ignoreTrailingSlash` server option is released as `v7.1.0`
Thanks for reporting this!

**Comment by Recodify** [comment]
This really should be enable by default
